### ISSUE TITLE
[FR]: Migrate project to testFixtures

### ISSUE THREAD

**Issue by SimonMarquis** [android/nowinandroid#451]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the problem

Even though [test fixtures](https://docs.gradle.org/current/userguide/java_testing.html) are not completely supported by the Android Gradle Plugin (specifically with Kotlin sources) we can still benefit a lot from this feature on JVM modules, and this should definitely be part of Android dev's best practices!

### Describe the solution

This would remove **a lot** of code duplication and prevent unecessary hacks like shared `sourcesSet` and gradle fixtures `module`s that could impact build times, especially in large code bases.
In the current state of the code base, it sheds ~1000 lines of code.

At my current company, we chose to use `kotlin.Random` as receiver for these testFixtures to make them more explicit about their behavior (being random by default), but this can be changed here if needed.

Next steps will be to migrate:
- "test" modules: `:core:data-test`, `:core:datastore-test`, `:sync-test`
- fakes and test classes like `FakeStringDecoder`, `TestAuthorsRepository`, `TestNewsRepository`, `TestTopicsRepository`, `TestUserDataRepository`

These next steps could already be done, but would need to rewrite classes in Java to be visible, and I don't think this is a path we want to go for this project.
So I guess we'll wait for the full Android/Kotlin support.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JolandaVerhoef** [comment]
This looks like an interesting addition, however I'd like to wait for support by Jetbrains (through https://youtrack.jetbrains.com/issue/KT-50667) before adding this to the project.

**Comment by SimonMarquis** [comment]
Just to clarify, testFixture _are_ officially supported.
The only bit that is not yet supported is exposing Kotlin sources from an AGP (`com.android.application` or `com.android.library`) module.
But I can understand if we don't want to include this feature until it is completely supported 👍

**Comment by yogurtearl** [comment]
Blocked by this issue: https://issuetracker.google.com/issues/259523353

**Comment by SimonMarquis** [comment]
For anyone interested in updates, I've pushed a new PR that enables "almost" proper Android test fixtures thanks to AGP 8.5.0:
- #1515
